spec/unit/lib/temporal/workflow/dispatcher_spec.rb

```diff
@@ -43,20 +46,42 @@
       end
 
       it 'stores the target and handler once' do
-        expect(handlers[target]).to be_kind_of(Array)
+        expect(handlers[target]).to be_kind_of(Hash)
         expect(handlers[target].count).to eq 1
       end
 
       it 'associates the event name and handler name with the target' do
-        event = handlers[target].first
+        event = handlers[target][1]
         expect(event.event_name).to eq(event_name)
       end
 
       it 'associates the handler with the target' do
-        event = handlers[target].first
+        event = handlers[target][1]
         expect(event.handler).to eq(block)
       end
     end
+
+    it 'removes a given handler against the target' do
+      block1 = -> { 'handler body' }
+      block2 = -> { 'other handler body' }
+      block3 = -> { 'yet another handler body' }
+
+      handle1 = subject.register_handler(target, 'signaled', &block1)
+      subject.register_handler(target, 'signaled', &block2)
+      subject.register_handler(other_target, 'signaled', &block3)
+
+      expect(subject.send(:handlers)[target][1].event_name).to eq('signaled')
+      expect(subject.send(:handlers)[target][1].handler).to be(block1)
+
+      expect(subject.send(:handlers)[target][2].event_name).to eq('signaled')
+      expect(subject.send(:handlers)[target][2].handler).to be(block2)
+
+      expect(subject.send(:handlers)[other_target][3].event_name).to eq('signaled')
+      expect(subject.send(:handlers)[other_target][3].handler).to be(block3)
+
+      handle1.unregister
+      expect(subject.send(:handlers)[target][1]).to be(nil)
```

Comment:
nit: Might wanna check that the others are still there. If you empty the hash completely it won't fail this spec